include/envoy/common/platform.h

```diff
@@ -125,6 +125,20 @@ struct msghdr {
 #define ENVOY_SHUT_WR SD_SEND
 #define ENVOY_SHUT_RDWR SD_BOTH
 
+// Mapping winsock2 errors to common error names
+#define SOCKET_ERROR_AGAIN WSAEWOULDBLOCK
```

Comment:
I am wondering if we should document this decision a bit more inline, i.e., expand on the thinking about defining envoy-neutral socket errors and maping them to platform specific definitions?

Reply:
Added some more content here, lmk if this is sufficient

Reply:
awesome, this is great!